gptme/tools/__init__.py

```diff
@@ -120,7 +124,17 @@ def execute_msg(msg: Message, confirm: ConfirmFunc) -> Generator[Message, None,
 
     for tooluse in ToolUse.iter_from_content(msg.content):
         if tooluse.is_runnable:
-            yield from tooluse.execute(confirm)
+            try:
+                for tool_response in tooluse.execute(confirm):
+                    yield tool_response.replace(call_id=tooluse.call_id)
+            except KeyboardInterrupt:
+                clear_interruptible()
```

Comment:
I'm not sure this belongs so deep into the tools module, should probably be kept in `chat.py`

Reply:
Ah, now I see why.